framework/src/components/system/system.js

```diff
@@ -93,8 +93,8 @@ class System {
 		if (!protocolVersion) {
 			return false;
 		}
-		const peerHard = parseInt(protocolVersion[0]);
-		const myHard = parseInt(this.headers.protocolVersion[0]);
+		const peerHard = parseInt(protocolVersion.split('.')[0]);
```

Comment:
I think the name `peerHard` is not so clear, maybe we can have some name like `peerHardForks` or `numPeerHForks`. Whatever you feel could be better than `peerHard` is fine.

Reply:
Also, if `protocolVersion` is `undefined` then this line will throw an error, or we already make sure that this is never `undefined`?

Reply:
if the protocol version is undefined it's being checked for in line 93